### PR TITLE
DAT-3411 Scrollbox matcher and test

### PR DESCRIPTION
**Pull request by SebastianMarzjan**
It looks that vast majority is Scroll_box, with sporadic redirects from Scrollbox and sometimes rewritten as Scroll or Infoscroll.
I decided to exclude Scroll as it can cause a lot of false positives and redirects as ones we don't really classify based on their target

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 3 major
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/pandora/blob/0ff0fea796275203965054ed4cab2d75d9d9f531/workers/template-classification-worker/src/test/java/com/wikia/workers/templateclassificationworker/TemplateParserTest.java#L105) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/0ff0fea796275203965054ed4cab2d75d9d9f531/workers/template-classification-worker/src/test/java/com/wikia/workers/templateclassificationworker/TemplateParserTest.java#L24) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Define a constant instead of duplicating this literal "quote" 3 times.](https://github.com/Wikia/pandora/blob/0ff0fea796275203965054ed4cab2d75d9d9f531/workers/template-classification-worker/src/test/java/com/wikia/workers/templateclassificationworker/TemplateParserTest.java#L65) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Either remove or fill this block of code.](https://github.com/Wikia/pandora/blob/0ff0fea796275203965054ed4cab2d75d9d9f531/workers/template-classification-worker/src/test/java/com/wikia/workers/templateclassificationworker/TemplateParserTest.java#L105) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00108)
* MINOR [Define a constant instead of duplicating this literal "navbox" 3 times.](https://github.com/Wikia/pandora/blob/0ff0fea796275203965054ed4cab2d75d9d9f531/workers/template-classification-worker/src/test/java/com/wikia/workers/templateclassificationworker/TemplateParserTest.java#L51) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)

File: workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/ScrollboxMatcher.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  /**
+   * @return name of matched type
+   */
+  public String getName() {
+    return "scrollbox";
+  }
+
+  /**
+   * check if title of document contains string "quot".
```
Comment (idradm) [inline workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/ScrollboxMatcher.java]: copypasting :warning:

File: workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/ScrollboxMatcher.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    return !document.select("title:matchesOwn(.+:Infoscroll|Scroll_box.*)").isEmpty();
+  }
+
+  /**
+   * Check if we match Wikipedia's example code
+   *
+   * @param document template's HTML
+   * @return boolean
+   */
+  private boolean matchStandardMarkup(Document document) {
```
Comment (idradm) [inline workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/ScrollboxMatcher.java]: hmmm... is there any reason for having this?

Comment (SebastianMarzjan) [inline workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/ScrollboxMatcher.java]: None. Good catch!

File: workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/ScrollboxMatcher.java (hunk 1 above)
Comment (SebastianMarzjan) [inline workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/ScrollboxMatcher.java]: :see_no_evil: 
will fix

File: workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/ScrollboxMatcher.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import com.wikia.workers.templateclassificationworker.TemplateTypeMatcher;
+
+import org.jsoup.nodes.Document;
+
+public class ScrollboxMatcher implements TemplateTypeMatcher {
+
+  public Boolean match(Document document) {
+    return
+        this.matchTitle(document);
```
Comment (idradm) [inline workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/ScrollboxMatcher.java]: This hurts my inner aesthetics.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Either remove or fill this block of code.](https://github.com/Wikia/pandora/blob/1a93714159e9083ee878f0a3ea7f11bfc932f422/workers/template-classification-worker/src/test/java/com/wikia/workers/templateclassificationworker/TemplateParserTest.java#L105) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00108)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/1a93714159e9083ee878f0a3ea7f11bfc932f422/workers/template-classification-worker/src/test/java/com/wikia/workers/templateclassificationworker/TemplateParserTest.java#L24) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)

**Comment by idradm** [comment]
LGTM

File: workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/ScrollboxMatcher.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package com.wikia.workers.templateclassificationworker.matchers;
+
+import com.wikia.workers.templateclassificationworker.TemplateTypeMatcher;
+
+import org.jsoup.nodes.Document;
+
+import java.lang.Override;
```
Comment (WikiaTech) [inline workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/ScrollboxMatcher.java]: MINOR Remove this unnecessary import: java.lang classes are always implicitly imported. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

File: workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/QuoteMatcher.java [hunk 5]
```diff
@@ -4,8 +4,11 @@
 
 import org.jsoup.nodes.Document;
 
+import java.lang.Override;
```
Comment (WikiaTech) [inline workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/QuoteMatcher.java]: MINOR Remove this unnecessary import: java.lang classes are always implicitly imported. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* MAJOR 4 major
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Move the "table" string literal on the left side of this string comparison.](https://github.com/Wikia/pandora/blob/3f7e810b1031fd6cf89b2b437fc4378671002fad/workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/NavboxMatcher.java#L93) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)
* MAJOR [Move the "div" string literal on the left side of this string comparison.](https://github.com/Wikia/pandora/blob/3f7e810b1031fd6cf89b2b437fc4378671002fad/workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/NavboxMatcher.java#L93) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)
* MAJOR [Either remove or fill this block of code.](https://github.com/Wikia/pandora/blob/3f7e810b1031fd6cf89b2b437fc4378671002fad/workers/template-classification-worker/src/test/java/com/wikia/workers/templateclassificationworker/TemplateParserTest.java#L105) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00108)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/3f7e810b1031fd6cf89b2b437fc4378671002fad/workers/template-classification-worker/src/test/java/com/wikia/workers/templateclassificationworker/TemplateParserTest.java#L24) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)

File: workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/ScrollboxMatcher.java (hunk 4 above)
Comment (idradm) [inline workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/ScrollboxMatcher.java]: :+1:

File: workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/QuoteMatcher.java (hunk 5 above)
Comment (idradm) [inline workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/QuoteMatcher.java]: :+1:

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 2 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Move the "table" string literal on the left side of this string comparison.](https://github.com/Wikia/pandora/blob/e699c1e518b2b29e9f3541606de1e8536b9f4444/workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/NavboxMatcher.java#L93) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)
* MAJOR [Move the "div" string literal on the left side of this string comparison.](https://github.com/Wikia/pandora/blob/e699c1e518b2b29e9f3541606de1e8536b9f4444/workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/NavboxMatcher.java#L93) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)

**Comment by idradm** [comment]
LGTM, could you update it with main branch?

**Comment by SebastianMarzjan** [comment]
Moved to https://github.com/Wikia/pandora/pull/575

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 2 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Move the "table" string literal on the left side of this string comparison.](https://github.com/Wikia/pandora/blob/e699c1e518b2b29e9f3541606de1e8536b9f4444/workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/NavboxMatcher.java#L93) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)
* MAJOR [Move the "div" string literal on the left side of this string comparison.](https://github.com/Wikia/pandora/blob/e699c1e518b2b29e9f3541606de1e8536b9f4444/workers/template-classification-worker/src/main/java/com/wikia/workers/templateclassificationworker/matchers/NavboxMatcher.java#L93) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)

**Pull request closed without merging**